File: docs/breez-sdk/snippets/wasm/receive.ts
import type { BreezSdk } from '@breeztech/breez-sdk-spark'

// ANCHOR: create-invoice
const createInvoiceExample = async (sdk: BreezSdk): Promise<void> => {
  // Create a Lightning invoice for 1000 sats
  const result = await sdk.createInvoice({
    amountSats: 1000,
    description: 'Coffee payment',
    expirySecs: 3600
  })
  console.log(`Invoice: ${result.bolt11}`)
  console.log(`Fee: ${result.feeSats} sats`)
}
// ANCHOR_END: create-invoice

// ANCHOR: create-spark-invoice
const createSparkInvoiceExample = async (sdk: BreezSdk): Promise<void> => {
  // Create a Spark invoice for 500 sats
  const result = await sdk.createSparkInvoice({
    amount: '500',
    description: 'Spark payment'
  })
  console.log(`Spark invoice: ${result.invoice}`)
  console.log(`Fee: ${result.fee}`)
}
// ANCHOR_END: create-spark-invoice

// ANCHOR: get-bitcoin-address
const getBitcoinAddressExample = async (sdk: BreezSdk): Promise<void> => {
  const result = await sdk.getBitcoinAddress()
  console.log(`Deposit address: ${result.address}`)
}
// ANCHOR_END: get-bitcoin-address

// ANCHOR: get-spark-address
const getSparkAddressExample = async (sdk: BreezSdk): Promise<void> => {
  const result = await sdk.getSparkAddress()
  console.log(`Spark address: ${result.address}`)
}
// ANCHOR_END: get-spark-address

export {
  createInvoiceExample,
  createSparkInvoiceExample,
  getBitcoinAddressExample,
  getSparkAddressExample
}

File: docs/breez-sdk/snippets/rust/src/receive.rs
use anyhow::Result;
use breez_sdk_spark::*;

// ANCHOR: create-invoice
async fn create_invoice_example(sdk: &BreezSdk) -> Result<()> {
    // Create a Lightning invoice for 1000 sats
    let result = sdk
        .create_invoice(Some(InvoiceOptions {
            amount_sats: Some(1000),
            description: Some("Coffee payment".to_string()),
            expiry_secs: Some(3600),
        }))
        .await?;
    println!("Invoice: {}", result.bolt11);
    println!("Fee: {} sats", result.fee_sats);
    Ok(())
}
// ANCHOR_END: create-invoice

// ANCHOR: create-spark-invoice
async fn create_spark_invoice_example(sdk: &BreezSdk) -> Result<()> {
    // Create a Spark invoice for 500 sats
    let result = sdk
        .create_spark_invoice(Some(SparkInvoiceOptions {
            amount: Some(500),
            description: Some("Spark payment".to_string()),
            ..Default::default()
        }))
        .await?;
    println!("Spark invoice: {}", result.invoice);
    println!("Fee: {}", result.fee);
    Ok(())
}
// ANCHOR_END: create-spark-invoice

// ANCHOR: get-bitcoin-address
async fn get_bitcoin_address_example(sdk: &BreezSdk) -> Result<()> {
    let result = sdk.get_bitcoin_address().await?;
    println!("Deposit address: {}", result.address);
    Ok(())
}
// ANCHOR_END: get-bitcoin-address

// ANCHOR: get-spark-address
async fn get_spark_address_example(sdk: &BreezSdk) -> Result<()> {
    let result = sdk.get_spark_address().await?;
    println!("Spark address: {}", result.address);
    Ok(())
}
// ANCHOR_END: get-spark-address
